x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -389,14 +409,46 @@ private <T extends TransportResponse> void sendWithRemoteAccessHeaders(
                             // authentication instance
                             new RemoteAccessAuthentication(authentication, RoleDescriptorsIntersection.EMPTY)
                         ).writeToContext(threadContext);
-                        sender.sendRequest(connection, action, request, options, contextRestoreHandler);
+                        final Transport.Connection effectiveConnection;
+                        final String effectiveAction;
+                        final TransportRequest effectiveRequest;
+                        if (false == TransportService.HANDSHAKE_ACTION_NAME.equals(action) && action.startsWith("internal:")) {
+                            effectiveConnection = TransportService.unwrapConnection(connection);
+                            final boolean isProxyConnection = effectiveConnection != connection;
+                            final DiscoveryNode targetNode = isProxyConnection ? connection.getNode() : null;
+                            effectiveAction = RemoteClusterActionProxy.getProxyAction(action);
+                            effectiveRequest = RemoteClusterActionProxy.wrapRequest(targetNode, request);
```

Comment:
Proxy actions here instead of `InternalRemoteConnection` because these proxies are only needed when security is enabled. When security is enabled, it mandates authorization and cluster privileges do not work with internal actions. It is not an issue at all when security is off. Hence it is better to stay within security area.